WordPress/Classes/ViewRelated/Reader/Comments/ReaderCommentsViewController.m

```diff
@@ -848,33 +852,40 @@ - (NSAttributedString *)cacheContentForComment:(Comment *)comment
 /// method locates that comment and scrolls the tableview to display it.
 - (void)navigateToCommentIDIfNeeded
 {
-    if (self.navigateToCommentID != nil) {
-        // Find the comment if it exists
-        NSArray<Comment *> *comments = [self.tableViewHandler.resultsController fetchedObjects];
-        NSArray<Comment *> *filteredComments = [comments filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"commentID == %@", self.navigateToCommentID]];
-        Comment *comment = [filteredComments firstObject];
+    if (self.navigateToCommentID == nil) {
+        return;
+    }
+    double delayInSeconds = 0.5;
+    dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(delayInSeconds * NSEC_PER_SEC));
+    dispatch_after(popTime, dispatch_get_main_queue(), ^(void){
+        [self scrollToCommentID];
+    });
+}
 
-        if (!comment) {
-            return;
-        }
+- (void)scrollToCommentID
+{
+    // Find the comment if it exists
+    NSArray<Comment *> *comments = [self.tableViewHandler.resultsController fetchedObjects];
+    NSArray<Comment *> *filteredComments = [comments filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"commentID == %@", self.navigateToCommentID]];
+    Comment *comment = [filteredComments firstObject];
 
-        // Force the table view to be laid out first before scrolling to indexPath.
-        // This avoids a case where a cell instance could be orphaned and displayed randomly on top of the other cells.
-        NSIndexPath *indexPath = [self.tableViewHandler.resultsController indexPathForObject:comment];
-        [self.tableView layoutIfNeeded];
-
-        // Ensure that the indexPath exists before scrolling to it.
-        if (indexPath.section >=0
-            && indexPath.row >=0
-            && indexPath.section < self.tableView.numberOfSections
-            && indexPath.row < [self.tableView numberOfRowsInSection:indexPath.section])
-        {
-            [self.tableView scrollToRowAtIndexPath:indexPath atScrollPosition:UITableViewScrollPositionTop animated:YES];
-            self.highlightedIndexPath = indexPath;
-        }
+    if (!comment) {
+        return;
+    }
 
-        // Reset the commentID so we don't do this again.
-        self.navigateToCommentID = nil;
```

Comment:
[Question] Was this removed by design? You mentioned a regression related to scrolling more than once to the comment. Could this be the cause of the regression?

Reply:
Yes, this change was intentional. Basically, it was the initial fix for my previous PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/23255
It occurs that the issue is complex, and `self.navigateToCommentID = nil;` breaks proper scrolling to the comment in case there are more pages to load.

Reply:
> Could this be the cause of the regression?
Exactly, that was the cause of the regression. We refresh the list (which triggers auto-scroll) when replying to a comment or a post. I've added an argument to the `refreshTableViewAndNoResultsView` method to manually control the auto-scroll.